### PR TITLE
Schedule principal cleanup job using UMB

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-31961

## Description of Intent of Change(s)
Schedule jobs based on PRINCIPAL_CLEANUP_DELETION_ENABLED_UMB. Do not schedule jobs to do synchronize clean up when it is true. Only schedule jobs that use UMB when enabled.

## Local Testing
Unit tests all pass;
Podman compose starting the scheduler based on the PRINCIPAL_CLEANUP_DELETION_ENABLED_UMB works as expected

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: deploy/rbac-clowdapp.yml
Comment (abaiken) [inline deploy/rbac-clowdapp.yml:3469]: should this default be true since this is setting it for commercial? Or do you prefer to leave it false here and set it to true in app interface?

**Review by abaiken — COMMENTED** [review]
This looks good to me & tracks with my thinking! Left one question but I'm happy to approve 👍

File: deploy/rbac-clowdapp.yml
Comment (astrozzc) [reply deploy/rbac-clowdapp.yml:3469]: Yea, good question! 
I set it to true for stage and prod in app-interface, which are the only places that can use UMB. So we don't have to set it in FedRamp (which we don't have access)

And this needs your approval, which can go before this PR: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/103147

Comment (abaiken) [reply deploy/rbac-clowdapp.yml:3469]: Awesome!! That was my next question - to make sure we set that before this was merged 👍 I LGTM the app-interface MR

**Review by abaiken — APPROVED** [review]

**Review by petracihalova — APPROVED** [review]
nice job 👍

**Pull request merged**
